File: solutions/elixir/allergies/1/lib/allergies.ex
defmodule Allergies do
  @points %{
    "eggs" => 1,
    "peanuts" => 2,
    "shellfish" => 4,
    "strawberries" => 8,
    "tomatoes" => 16,
    "chocolate" => 32,
    "pollen" => 64,
    "cats" => 128
  }

  def list(flags) do
    @points
    |> Map.keys()
    |> Enum.filter(&allergic_to?(flags, &1))
  end

  def allergic_to?(flags, item) do
    Bitwise.band(flags, @points[item]) != 0
  end
end

File: solutions/fsharp/allergies/1/Allergies.fs
module Allergies

open System

type Allergen =
    | Eggs = 1
    | Peanuts = 2
    | Shellfish = 4
    | Strawberries = 8
    | Tomatoes = 16
    | Chocolate = 32
    | Pollen = 64
    | Cats = 128

let allergicTo (codedAllergies: int) (allergen: Allergen) : bool = 
    codedAllergies &&& int allergen <> 0

let list codedAllergies =
    Enum.GetValues typeof<Allergen>
    |> Seq.cast<Allergen>
    |> Seq.filter (fun a -> allergicTo codedAllergies a)
    |> Seq.toList
